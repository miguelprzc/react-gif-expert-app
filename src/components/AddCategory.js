import React, { useState } from 'react'
import { PropTypes } from 'prop-types';

export const AddCategory = ({setCategories}) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    if(inputValue.trim()) {
      setCategories(cats => [inputValue, ...cats]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input 
        type="text"
        placeholder="What are you looking for?"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
