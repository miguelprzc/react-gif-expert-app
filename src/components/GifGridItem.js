import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {
  return (
    <figure className=" card animate__animated animate__fadeInUp">
      <img src={url} alt={title} />
      <figcaption>{title}</figcaption>
    </figure>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
