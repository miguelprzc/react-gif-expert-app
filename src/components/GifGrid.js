import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

  const {data: gifs, loading} = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeInUp">{category}</h3>
      <div className="card-grid">
        {
          loading && <p className="animate__animated animate__flash">Loading...</p>
        }

        {
          gifs.map((gif) => (
            <GifGridItem 
              key={gif.id} 
              {...gif} 
            />
          ))
        }
      </div>
    </>
  )
}
