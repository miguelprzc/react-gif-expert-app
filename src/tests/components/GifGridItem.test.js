import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('GifGridItem', () => {

  const title = "A title";
  const url = "https://localhost/something.gif";

  const wrapper = shallow(<GifGridItem title={title} url={url}/>);

  test('should show <GifGridItem /> properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should have a figcaption with the title', () => {
    const figcaption = wrapper.find('figcaption');
    
    expect(figcaption.text().trim()).toBe(title); 
  });

  test('should have an img with the url as url and the alt as title', () => {
    const img = wrapper.find('img');
    const {src, alt} = img.props();
    
    expect(src).toBe(url); 
    expect(alt).toBe(title); 
  });

  test('should have a animate_fadeInUp as a css class', () => {
    const figure = wrapper.find('figure');
    const {className} = figure.props();

    expect(className.includes('animate__fadeInUp')).toBeTruthy();
  });
  
  

});

