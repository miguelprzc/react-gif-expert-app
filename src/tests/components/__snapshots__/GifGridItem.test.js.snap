// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem should show <GifGridItem /> properly 1`] = `
<figure
  className=" card animate__animated animate__fadeInUp"
>
  <img
    alt="A title"
    src="https://localhost/something.gif"
  />
  <figcaption>
    A title
  </figcaption>
</figure>
`;
